### DISCUSSION TITLE
How to publish HTML file?

### DISCUSSION THREAD

**Discussion by vitalyk-multinarity** [community/community#38559] [Actions] [answered]
It seems to be a very trivial task, but I'm stuck.  I have an HTML report (produced on some step) and I want to publish it. How should I do that? 
TIA,
Vitaly

**Comment by david-engelmann** [comment]
You can open local HTML in your web browser for demoing purposes

**Comment by subidit** [comment]
If you want to show the `HTML` rendered page then just open the file in a browser. 
Or if you want to show the `HTML` code then open the file in a code editor like Notepad or VS Code.
you could right click on the file, then you could choose the desired application to open your file in.

**Comment by ihnatenkoo** [comment]
Привет, можешь подробней описать проблему

**Comment by airtower-luna** [comment, marked as answer]
If you mean that you want to show the HTML file directly in the web UI for an Actions job/workflow, you can't. What you can do:

* You can create a [job summary](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#adding-a-job-summary) in Markdown rather than HTML. That will be shown on the workflow page.
* You can upload the HTML file [as an artifact](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts). To see it you'll have to download the artifact and open the file locally.

**Reply by vitalyk-multinarity** [reply]
>If you mean that you want to show the HTML file directly in the web UI for an Actions job/workflow,
@airtower-luna - thank you, it was exactly my question. 
@ihnatenkoo @subidit @david-engelmann  -  thanks and sorry, my question wasn't clear enough.

[In fact, meanwhile I figured out that for both 'job summary' and 'report' I need markdown, not HTML, so I need to convert. For some reason, 'pandoc' produces broken .md, and rknj/html2markdown's  markdown is almost OK but without colors. In the worst case, I'll just upload HTML as an artifact.]
